src/sulogin-shell/sulogin-shell.c

```diff
@@ -41,6 +43,32 @@ static int reload_manager(sd_bus *bus) {
         return 0;
 }
 
+static int is_inactive_default_target(sd_bus *bus, bool *ret_is_inactive) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_free_ char *path = NULL, *state = NULL;
+        int r = 0;
+
+        path = unit_dbus_path_from_name(SPECIAL_DEFAULT_TARGET);
+        if (!path) {
+                log_warning_errno(ENOMEM, "Failed to allocate a dbus path for "SPECIAL_DEFAULT_TARGET": %m");
+                return -ENOMEM;
+        }
+
+        r = sd_bus_get_property_string(bus,
+                                       "org.freedesktop.systemd1",
+                                       path,
+                                       "org.freedesktop.systemd1.Unit",
+                                       "ActiveState",
+                                       &error,
+                                       &state);
+        if (r < 0)
+                return log_error_errno(r, "Failed to retrieve unit state: %s", bus_error_message(&error, r));
+
+        *ret_is_inactive = streq_ptr(state, "inactive");
+
+        return 0;
```

Comment:
I think it is sufficient to return the result through `return`:
```c
return !streq_ptr(state, "inactive");
```

Also, maybe,
```c
return STR_IN_SET(state, "active", "activating");
```
is better? That is, what should we do the unit is e.g. in `"failed"` state?

Reply:
Yes, what I'm saying in the previous comment is about this. is_active() doesn't correspond with what it is actually doing if it returns true for the states other than active state.

It would be helpful if any comment is given on how to handle other states. I'm choosing inactive only and this is very conservative setting.